frontend/src/utils/excelParser.ts

```diff
@@ -0,0 +1,250 @@
+import XLSX from 'xlsx';
+import {
+    Schedule,
+    ScheduleCourse,
+    SeasonEnum,
+    ScheduleYear,
+    ScheduleTerm,
+    StatusEnum
+} from "../models/types";
+import { convertSeasonToTermId } from './schedule-helpers';
+
+const BASE_YEAR: number = 1000;
+
+export function ExcelToSchedule(file: File, callback: ((schedule: Schedule) => any)) {
```

Comment:
I thought this was a class name because of the capitalization, I think `excelToSchedule` would be more clear